lib/transition/unpack.c

```diff
@@ -164,6 +164,130 @@ unpack_synapse(crm_graph_t * new_graph, xmlNode * xml_synapse)
     return new_synapse;
 }
 
+static xmlNode *
+get_start_action_xml(synapse_t *synapse)
+{
+    GListPtr lpc;
+
+    for (lpc = synapse->actions; lpc != NULL; lpc = lpc->next) {
+        crm_action_t *action = (crm_action_t *) lpc->data;
+        const char *operation = crm_element_value(action->xml, XML_LRM_ATTR_TASK);
+
+        if (action->type == action_type_rsc && safe_str_eq(operation, RSC_START)) {
+            return action->xml;
+        }
+    }
+
+    return NULL;
+}
+
+static int
+get_max_action_id(crm_graph_t *graph)
+{
+    GListPtr sIter;
+    int id = 0;
+
+    for (sIter = graph->synapses; sIter != NULL; sIter = sIter->next) {
+        synapse_t *synapse = (synapse_t *) sIter->data;
+        GListPtr aIter;
+
+        for (aIter = synapse->actions; aIter != NULL; aIter = aIter->next) {
+            crm_action_t *action = (crm_action_t *) aIter->data;
+
+            if (action->id > id) {
+                id = action->id;
+            }
+        }
+    }
+
+    return id;
+}
+
+static void destroy_action(crm_action_t * action);
+
+static gboolean
+inject_metadata(crm_graph_t *graph, synapse_t *parent_synapse)
+{
+    crm_action_t *new_input = NULL;
+    synapse_t *new_synapse = NULL;
+    xmlNode *synapse = NULL;
+    xmlNode *action_set = NULL;
+    xmlNode *op = NULL;
+    xmlNode *rsc = NULL;
+    xmlNode *attrs = NULL;
+    char *op_key = NULL;
+    xmlNode *start = NULL;
+    xmlNode *start_rsc = NULL;
+    xmlNode *start_attrs = NULL;
+
+    if (!graph || !parent_synapse) {
+        return FALSE;
+    }
+
+    start = get_start_action_xml(parent_synapse);
+    if (!start) {
+        return FALSE;
+    }
```

Comment:
We don't really need to know ra-versions when the probe is done since it doesn't have any versioned attributes. The stop operations, in turn, don't evaluate any parameters, they use the parameters the resource started with. So, I think having metadata call before the start is enough, isn't it?

We probably could have these calls injected only once for each agent on the same node per transition, right.